gril/grilunsol.c

```diff
@@ -575,7 +585,19 @@ int g_ril_unsol_parse_signal_strength(GRil *gril, const struct ril_msg *message,
 		parcel_r_int32(&rilp); /* rsrq */
 		lte_rssnr = parcel_r_int32(&rilp);
 		parcel_r_int32(&rilp); /* cqi */
+		/* Samsung on MSM822x calculation of signal strength,
+		according to LineageOS's RIL adaptor */
+		if (samsung_quirks) {
+			if ((lte_sigstr & 0xff) == 255 || lte_sigstr == 99) {
+				lte_sigstr = 99;
+				lte_rsrp = INT_MAX;
+				lte_rssnr = INT_MAX;
+			} else {
+				lte_sigstr &= 0xff;
+			}
+		}
 		lte_signal = get_lte_strength(lte_sigstr, lte_rsrp, lte_rssnr);
+		parcel_r_int32(&rilp); /* tdScdmaRscp */
```

Comment:
This was never read and I suggest to read it off the parcel, just for completeness.

Reply:
This field seems to be added since RIL version 10.
https://github.com/LineageOS/android_hardware_ril/blob/cm-14.1/include/telephony/ril.h#L1257

You'll have to properly guard this read under either `parcel_data_avail()` or RIL version check. In fact, in RIL version 8 there's a field between this and `tdScdmaRscp`: `lteTimingAdvance`. As this isn't required for the Samsung MSM822x devices, and this isn't a part of an "array", sets of the same data type repeating a number of times, I would say this should be removed.

Reply:
Removed!